src/Containers/packaging/build/Microsoft.NET.Build.Containers.targets

```diff
@@ -272,6 +300,92 @@
       <Output TaskParameter="GeneratedContainerDigest" PropertyName="GeneratedContainerDigest" />
       <Output TaskParameter="GeneratedArchiveOutputPath" PropertyName="GeneratedArchiveOutputPath" />
       <Output TaskParameter="GeneratedContainerNames" ItemName="GeneratedContainerName" />
+      <Output TaskParameter="GeneratedContainerArchitecture" PropertyName="GeneratedContainerArchitecture" />
+      <Output TaskParameter="GeneratedContainerOS" PropertyName="GeneratedContainerOS" />
     </CreateNewImage>
+
+    <ItemGroup>
+      <GeneratedManifestInfo Include="ManifestInfo">
+        <ManifestLength>$(GeneratedContainerManifest.Length)</ManifestLength>
+        <Digest>$(GeneratedContainerDigest)</Digest>
+        <Architecture>$(GeneratedContainerArchitecture)</Architecture>
+        <OS>$(GeneratedContainerOS)</OS>
+      </GeneratedManifestInfo>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_PublishMultiArchContainers" DependsOnTargets="$(PublishContainerDependsOn)" >
+    <Error Text="ContainerArchiveOutputPath is invalid. It can only be a directory in multi-arch scenario because the runtime identifier is appended in the end of the filename."
+      Condition="'$(ContainerArchiveOutputPath)' != '' and 
+                 !$(ContainerArchiveOutputPath.EndsWith('/')) and 
+                 !$(ContainerArchiveOutputPath.EndsWith('\\')) and 
+                 $(ContainerArchiveOutputPath.EndsWith('.tar.gz'))" />
+    
+    <ItemGroup>
+      <_rids Include="$(ContainerRuntimeIdentifiers)" Condition="'$(ContainerRuntimeIdentifiers)' != ''" />
+      <_rids Include="$(RuntimeIdentifiers)" Condition="'$(ContainerRuntimeIdentifiers)' == '' and '$(RuntimeIdentifiers)' != ''" />
+      <_InnerBuild
+        Include="$(MSBuildProjectFullPath)"
+        AdditionalProperties="
+          ContainerRuntimeIdentifier=%(_rids.Identity);
+          ContainerBaseRegistry=$(ContainerBaseRegistry);
+          ContainerBaseName=$(ContainerBaseName);
+          ContainerBaseTag=$(ContainerBaseTag);
+          ContainerRegistry=$(ContainerRegistry);
+          _ContainerImageTags=@(ContainerImageTags, ';');
+          ContainerRepository=$(ContainerRepository);
+          ContainerWorkingDirectory=$(ContainerWorkingDirectory);
+          _ContainerEntrypoint=@(ContainerEntrypoint, ';');
+          _ContainerEntrypointArgs=@(ContainerEntrypointArgs, ';');
+          _ContainerAppCommand=@(ContainerAppCommand, ';');
+          _ContainerAppCommandArgs=@(ContainerAppCommandArgs, ';');
+          ContainerAppCommandInstruction=$(ContainerAppCommandInstruction);
+          _ContainerDefaultArgs=@(ContainerDefaultArgs, ';');
+          _ContainerLabel=@(ContainerLabel, ';');
+          _ContainerPort=@(ContainerPort, ';');
+          _ContainerEnvironmentVariables=@(ContainerEnvironmentVariables, ';');
```

Comment:
These three Items can't be 'passed' into the inner build like this because they contain metadata that gets removed when they are converted back to properties. As a result, when parsed back into Items during `_ParseParametersForPublishingSingleContainer` the Items exist but no longer have any of their useful values.

@rainersigwald is there a better pattern for computing properties and Items that should be provided to an Inner build?

Reply:
There's not really a good "push" model for items with metadata. You can "pull" via the return value of a target called with MSBuild; that might be an option here (to have the inner builds call back to the outer build to get the info). Otherwise you have to pass structured data and deserialize it on the other side (JSON or something).

Reply:
this is an interesting question though - you could see for some of these (env var/arguments) might need to vary per-RID.